src/dsp_tools/utils/xml_validation.py

```diff
@@ -59,25 +68,16 @@ def validate_xml(input_file: Union[str, Path, etree._ElementTree[Any]]) -> bool:
 
 def _parse_schema_and_data_files(
     input_file: Union[str, Path, etree._ElementTree[Any]],
-) -> tuple[Union[etree._ElementTree[etree._Element], etree._Element], etree.XMLSchema]:
+) -> tuple[etree._Element, etree.XMLSchema]:
     with importlib.resources.files("dsp_tools").joinpath("resources/schema/data.xsd").open(
         encoding="utf-8"
     ) as schema_file:
         xmlschema = etree.XMLSchema(etree.parse(schema_file))
-    if isinstance(input_file, (str, Path)):
-        try:
-            data_xml = etree.parse(source=input_file)
-        except etree.XMLSyntaxError as err:
-            logger.error(f"The XML file contains the following syntax error: {err.msg}", exc_info=True)
-            raise InputError(f"The XML file contains the following syntax error: {err.msg}") from None
-    else:
-        data_xml = input_file
+    data_xml = parse_and_remove_comments_xml_file(input_file)
```

Comment:
I feel you, and I have done such things in the past myself, and I had to make the painful experience that mostly, you don't do yourself a favor. Because if the other PR is modified (as it is the case now), you have to migrate the  modifications. Plus, for the reviewer, it can be really confusing, because multiple things are mixed together.

But it's fine for now. I have approved the other PR. As soon as you have merged it and "tidied up" this one, I can give you an approve.

Reply:
No worries, I tend not to do that either but as you both are not officially working, I did not wait for an approve. I do appreciate that you take a look even though you are ill. Otherwise the mess would be even greater.